src/agent/at-driver.js

```diff
@@ -62,6 +69,7 @@ export class ATDriver {
     for await (const rawMessage of iterateEmitter(this.socket, 'message', 'close', 'error')) {
       const message = rawMessage.toString();
       this.log(AgentMessage.AT_DRIVER_COMMS, { direction: 'inbound', message });
+      if (this.hasClosed) throw new Error('AT-Driver connection unexpectedly closed');
```

Comment:
The design of `iterateEmitter` suggests that this is not an exceptional circumstance--the utility seems designed to gracefully handle the case where the `completEvent` (which is `"close"` in this case) occurs while values are still being emitted.

I honestly don't have any recollection of the thinking that went into this, but from reviewing [the source](https://github.com/w3c/aria-at-automation-harness/blob/3daf7e0c65aee1ec0fb87d22ed070a11a2bc4128/src/shared/iterate-emitter.js), I believe it may be because values are always emitted asynchronously. It seems possible that `close` occurs while the utility is still working through its internal backlog of values.

If `"close"` *is* an exceptional circumstance in this context, then maybe we should specify `null` as the `completeEvent` for this invocation.

What do you think?

Reply:
hrm, you seem right, 'close' and 'error' are called out here, going to spend a little more time looking one place above these calls to handled unexpected error rather than this particular throw

Reply:
Thought this one through a lot and figured out where the throw needs to happen.

The `iterateEmitter` will properly throw if the connection is closed while it's iterating, but if it were to call into this `_messages` while starting in a closed state, it would never throw again.  Therefore I moved the check for an already closed socket to the top of _messages